mlir-tensorrt/compiler/lib/Conversion/TensorRTToTensorRTRuntime/TensorRTToTensorRTRuntime.cpp

```diff
@@ -72,6 +146,8 @@ class ConvertTensorRTToRuntimePass
     }
 
     IRRewriter rewriter(ctx);
+    SymbolTableCollection symbolTable;
+    DataFlowSolver solver;
```

Comment:
You're doing something different there, keeping a single solver and repeatedly re-initializing it? I've never seen that pattern before, not sure if it will actually work as intended.

Reply:
I did not realized that we need should not keep a single solver while iterating through different operations. I have reverted my change.

Reply:
Yeah what needs to be done to keep a solver is to initialize it once. Then when you do replace/rewrite, you need to make corresponding small adjustments/remaps to the solver state. We do this in the Plan dialect transforms, and ideally we can do that here as well, but it's not especially high priority.